h2o-genmodel/src/main/java/hex/genmodel/algos/gam/GamMojoModelBase.java

```diff
@@ -134,29 +133,26 @@ int readCatVal(double data, int dataIndex) {
   
   // this method will add to each data row the expanded gam columns
   public double[] addExpandGamCols(double[] rawData, final RowData rowData) { // add all expanded gam columns here
-    // add expanded gam columns to rowData
-    int dataIndStart = _totFeatureSize-_numExpandedGamCols; // starting index to fill out the rawData
-    double[] dataWithGamifiedColumns = null;  // only allocate this when gamification of columns are needed.
-    for (int cind = 0; cind < _num_gam_columns; cind++) {
-      if (_bs[cind]==0) { // to generate basis function values for cubic regression spline
-        Object dataObject = rowData.get(_gam_columns[cind]);
-        double gam_col_data = Double.NaN;
-        if (dataObject == null) {  // gamified columns are already included
-          return rawData;
-        } else
-          gam_col_data = (dataObject instanceof String)?Double.parseDouble((String)dataObject):(double) dataObject;
-        GamUtilsCubicRegression.expandOneGamCol(gam_col_data, _binvD[cind], _basisVals[cind], _hj[cind], _knots[cind]);
-      } else {
-        throw new IllegalArgumentException("spline type not implemented!");
+      // add expanded gam columns to rowData
+      int dataIndStart = _totFeatureSize - _numExpandedGamCols; // starting index to fill out the rawData
```

Comment:
@pavel:  You will run into problem here with testJavaScoring.  However, this should work when you are doing normal scoring for python unit.  I am puzzled by why it is failing at both places.

Reply:
I was able to run the python test to completion successfully.  However, as I mentioned before, it is not going to work with testJavaScore because there will be a gam_column missing and in this case, it is just going to leave NaN in the gamified columns.  Later during scoring, it will then replace the NaN with column  means.  That is why the model prediction and mojo predictions are different.